server/web/main.go

```diff
@@ -21,29 +21,37 @@ const (
 	restrictedPath = "/restricted"
 )
 
+// Specification represents a map of enviornment variables
+type Specification struct {
+	UserPassword  string `envconfig:"USERPASSWD" required:"true"`
+	EncryptionKey string `split_words:"true" required:"true"`
+
+	MongodbURI string `default:"mongodb://127.0.0.1:27017/db" split_words:"true"`
+	Port       string `default:"8080"`
+	PublicHTML string `split_words:"true" default:"public"`
+}
+
 func main() {
-	userPasswd := os.Getenv("USERPASSWD")
-	if userPasswd == "" {
-		log.Fatal("USERPASSWD must be set.")
+	var spec Specification
+	err := envconfig.Process("", &spec)
+	if err != nil {
+		log.Fatal(err.Error())
 	}
-	key := []byte(os.Getenv("ENCRYPTION_KEY"))
+
```

Comment:
Log the environment variables/config?

fmt.Printf("%+v", spec) ?

Reply:
I forgot to log it.. dammit